tools/cbmc/proofs/DNS/DNSgetHostByName_cancel/DNSgetHostByName_cancel_harness.c

```diff
@@ -0,0 +1,46 @@
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "queue.h"
+#include "list.h"
+
+/* FreeRTOS+TCP includes. */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_DNS.h"
+#include "FreeRTOS_IP_Private.h"
+
+/* Here we initializa the callbacklist and add item in order to check the cancel function.
+The functions pvPortMalloc, xTaskCheckForTimeOut, xTaskResumeAll are considered to be correct.*/
+void vDNSInitialise( void );
+void vDNSSetCallBack( const char *pcHostName, void *pvSearchID, FOnDNSEvent pCallbackFunction, TickType_t xTimeout, TickType_t xIdentifier );
+
+void *pvPortMalloc(size_t xWantedSize) {
+	void *pvSize = malloc(xWantedSize);
+	return pvSize;
+}
+
+BaseType_t xTaskCheckForTimeOut( TimeOut_t * const pxTimeOut, TickType_t * const pxTicksToWait ) {
+	BaseType_t xTimeout;
+	return xTimeout;
+}
+
+BaseType_t xTaskResumeAll( void ) {
+	BaseType_t xRet;
+	return xRet;
+}
+
+void func(const char * pcHostName, void * pvSearchID, uint32_t ulIPAddress) {}
+
+
+void harness() {
+	if (ipconfigDNS_USE_CALLBACKS==1) {
+		vDNSInitialise();
+		void *pvSearchID = malloc(sizeof(size_t));
```

Comment:
This is a bit confusing...if the function is expecting a pointer to a `size_t`, please could you just do
```c
size_t pvSearchID;
...
vDNSSetCallBack(pcHostName, &pvSearchID, ...);
```